src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.ChangeNamespaceCodeAction.cs

```diff
@@ -0,0 +1,526 @@
+            /// that new import should be added to based on reference locations.
+            /// Depends on whether <paramref name="namespaceParts"/> is specified,
+            /// the fix would be:
+            ///     1. qualify the reference with new namespace and mark it for simplification, or
+            ///     2. find and mark the qualified reference for simplification.
+            /// </summary>
+            private async Task<(Document, ImmutableArray<SyntaxNode>)> FixReferencesAsync(
+                Document document, 
+                IAddImportsService addImportService, 
+                ISyncNamespaceService syncNamespaceService,
+                IEnumerable<ReferenceLocation> refLocations, 
+                ImmutableArray<string> namespaceParts,          // No namespace replacement if this is default.  
+                CancellationToken cancellationToken)
+            {
+                var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                var root = editor.OriginalRoot;
+                var containers = new HashSet<SyntaxNode>();
+
+                var generator = SyntaxGenerator.GetGenerator(document);
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var dummyImport = CreateImport(generator, "Dummy", withFormatterAnnotation: false);
+
+                foreach (var refLoc in refLocations)
+                {
+                    Debug.Assert(document.Id == refLoc.Document.Id);
+
+                    // Ignore references via alias. For simple cases where the alias is defined as the type we are interested,
+                    // it will be handled properly because it is one of the reference to the type symbol. Otherwise, we don't
+                    // attempt to make a potential fix, and user might end up with errors as a result.                    
+                    if (refLoc.Alias != null)
+                    {
+                        continue;
+                    }
+
+                    // Other documents in the solution might have changed after we calculated those ReferenceLocation, 
+                    // so we can't trust anything to be still up-to-date except their spans.
+
+                    // Get inner most node in case of type used as a base type. e.g.
+                    //
+                    //      public class Foo {}
+                    //      public class Bar : Foo {}
+                    //
+                    // For the reference to Foo where it is used as a base class, the BaseTypeSyntax and the TypeSyntax
+                    // have exact same span.
+
+                    var refNode = root.FindNode(refLoc.Location.SourceSpan, getInnermostNodeForTie: true);
+                    if (syncNamespaceService.TryGetReplacementReferenceSyntax(
+                        refNode, namespaceParts, syntaxFacts, out var oldNode, out var newNode))
```

Comment:
indentation is wonky here.  when looking at this code, it looks like this is a sibling expression (that would be part of a boolean expression).  this should be indented further to make it clear it's an sub argument.